Sources/Chat.swift

```diff
@@ -0,0 +1,254 @@
+  )
+
+  func wherePresent(
+    userId: String,
+    completion: ((Swift.Result<[String], Error>) -> Void)?
+  )
+
+  func isPresent(
+    userId: String,
+    channelId: String,
+    completion: ((Swift.Result<Bool, Error>) -> Void)?
+  )
+
+  func createChannel(
+    id: String,
+    name: String?,
+    description: String?,
+    custom: [String: JSONCodableScalar]?,
+    type: ChannelType?,
+    status: String?,
+    completion: ((Swift.Result<ChatChannelType, Error>) -> Void)?
+  )
+
+  func getChannel(
+    channelId: String,
+    completion: ((Swift.Result<ChatChannelType?, Error>) -> Void)?
+  )
+
+  func getChannels(
+    filter: String?,
+    sort: [PubNub.ObjectSortField],
+    limit: Int?,
+    page: PubNubHashedPage?,
+    completion: ((Swift.Result<(channels: [ChatChannelType], page: PubNubHashedPage?), Error>) -> Void)?
+  )
+
+  func updateChannel(
+    id: String,
+    name: String?,
+    custom: [String: JSONCodableScalar]?,
+    description: String?,
+    status: String?,
+    type: ChannelType?,
+    completion: ((Swift.Result<ChatChannelType, Error>) -> Void)?
+  )
+
+  func deleteChannel(
+    id: String,
+    soft: Bool,
+    completion: ((Swift.Result<ChatChannelType, Error>) -> Void)?
+  )
+
+  func forwardMessage(
+    message: ChatMessageType,
+    channelId: String,
+    completion: ((Swift.Result<Timetoken, Error>) -> Void)?
+  )
+
+  func whoIsPresent(
+    channelId: String,
+    completion: ((Swift.Result<[String], Error>) -> Void)?
+  )
+
+  func emitEvent<T: EventContent>(
+    channelId: String,
+    payload: T,
+    mergePayloadWith otherPayload: [String: JSONCodable]?,
+    completion: ((Swift.Result<Timetoken, Error>) -> Void)?
+  )
+
+  func createPublicConversation(
+    channelId: String?,
+    channelName: String?,
+    channelDescription: String?,
+    channelCustom: [String: JSONCodableScalar]?,
+    channelStatus: String?,
+    completion: ((Swift.Result<ChatChannelType, Error>) -> Void)?
+  )
+
+  func createDirectConversation(
+    invitedUser: UserImpl,
+    channelId: String?,
+    channelName: String?,
+    channelDescription: String?,
+    channelCustom: [String: JSONCodableScalar]?,
+    channelStatus: String?,
+    membershipCustom: [String: JSONCodableScalar]?,
+    completion: ((Swift.Result<CreateDirectConversationResult<ChatChannelType, ChatMembershipType>, Error>) -> Void)?
+  )
+
+  func createGroupConversation(
+    invitedUsers: [UserImpl],
+    channelId: String?,
+    channelName: String?,
+    channelDescription: String?,
+    channelCustom: [String: JSONCodableScalar]?,
+    channelStatus: String?,
+    membershipCustom: [String: JSONCodableScalar]?,
+    completion: ((Swift.Result<CreateGroupConversationResult<ChatChannelType, ChatMembershipType>, Error>) -> Void)?
+  )
+
+  func listenForEvents<T: EventContent>(
+    type: T.Type,
+    channelId: String,
+    customMethod: EmitEventMethod,
+    callback: @escaping ((EventWrapper<T>) -> Void)
```

Comment:
All of them are async. This closure is a bit different, because it's not optional so you have to add `@escaping` keyword. And for all other closures which are optional - they're `@escaping` by default so you don't have to add any prefix